Userland/Libraries/LibCore/Stream.cpp

```diff
@@ -73,6 +73,18 @@ ErrorOr<ByteBuffer> Stream::read_all_impl(size_t block_size, size_t expected_fil
     return data;
 }
 
+ErrorOr<void> Stream::discard(size_t discarded_bytes, size_t chunk_size)
+{
+    auto buffer = TRY(ByteBuffer::create_uninitialized(chunk_size));
```

Comment:
I already got a preliminary "maybe we can make some exceptions", but that was before I had uploaded the PR in question.

That said, I completely forgot to override this for `File`, we can just seek there instead of actually having to read stuff.

Also, I guess I should limit the chunk size to the number of discarded bytes if the latter is smaller, no need to allocate a whole 4k if we just want to discard a bit of padding.

Reply:
Can we just remove the chunk size parameter (no one uses it?) and use a `PAGE_SIZE` stack buffer instead?
```c++
ErrorOr<void> Stream::discard(size_t discarded_bytes)
{
    Array<u8, PAGE_SIZE> buffer;

    while (discarded_bytes > 0) {
        auto slice = TRY(read(buffer.span().slice(0, min(discarded_bytes, buffer.size()))));
        discarded_bytes -= slice.size();
    }

    return {};
}
```

Reply:
The idea behind the chunk size parameter was that users can affect how much data it reads at once, similar to the `read_all` function just above.

But until something needs that I guess a buffer on the stack doesn't sound too bad. :^)